be-python-flask/Jenkinsfile.template

```diff
@@ -30,49 +30,61 @@ odsPipeline(
     '*': 'dev'
   ]
 ) { context ->
+  stageTestSuite(context, context.nexusHostWithBasicAuth, context.nexusHost.tokenize('//')[1])
+  stageScanForSonarqube(context)
   stageBuild(context)
   stageStartOpenshiftBuild(context, [nexusHostWithBasicAuth: context.nexusHostWithBasicAuth, nexusHostWithoutScheme: context.nexusHost.tokenize('//')[1]])
   stageDeployToOpenshift(context)
 }
 
+def stageTestSuite(def context, def nexusHostWithBasicAuth, def nexusHostWithoutScheme) {
+  String testLocation = 'build/test-results/test'
+  String coverageLocation = 'build/test-results/coverage'
+
+  stage('Prepare Test Suite') {
+    sh """
+      virtualenv testsuite
+      . ./testsuite/bin/activate
+      pip install -i ${nexusHostWithBasicAuth}/repository/pypi-all/simple --trusted-host ${nexusHostWithoutScheme} -r tests/requirements.txt
```

Comment:
If the user starts adding requirements these may not be compatible. However pip install will not return with a non-zero exit code in these cases. See https://github.com/pypa/pip/issues/6969.
I think adding a pip check after pip installs should do the trick.

Reply:
pip check in place! also in the Dockerfile :)